bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SyncCursor.java

```diff
@@ -0,0 +1,112 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.bookie;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Handle the LastAddSynced entryId. As entries do not arrive to the Journal in a known order we have to handle
+ * increments of this pointer.
+ *
+ * @since 4.7
+ */
+public class SyncCursor {
+
+    private long minAddSynced = -1;
+    private final SortedMap<Long, Long> ranges = new TreeMap<>();
```

Comment:
@sijie you requested a change about using some Guava utility for this. I cannot find the comment. Could you please give me again your suggestion ?.
I will be happy to have a more efficient implementation